src/main/java/io/cdap/plugin/servicenow/source/ServiceNowSource.java

```diff
@@ -74,6 +74,8 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // Since we have validated all the properties, throw an exception if there are any errors in the collector.
     // This is to avoid adding same validation errors again in getSchema method call
     collector.getOrThrowException();
+    stageConfigurer.setOutputSchema(
+      ServiceNowInputFormat.fetchTableInfo(conf.getQueryMode(collector), conf).get(0).getSchema());
```

Comment:
Please check if this works with macros enabled. What happens if query mode, table name are macros?

Reply:
hi @rmstar  

No, the schema won't be generated if the query mode and table name are set as Macro.

If we set these fields as Macro, then the following error appears at the time of validation.
**Unsupported query mode value: null Supported modes are: Reporting, Table**

<img width="1779" alt="Validation Error when Query Mode and Table Name are set as Macro" src="https://user-images.githubusercontent.com/89522746/142404943-193f4a59-ea8a-4b88-befa-4c79e919cdf2.png">

Reply:
Yes, so you need to handle this case. We should skip validation in configurePipeline for fields that have macros

Reply:
The logic to skip validation in `configurePipeline` for fields that have macros already exists in the codebase. I've added the shouldGetSchema() method to check each mandatory field that can have Macro before setting the output schema. 171aa6eeb43b3413335edf8505017cd5ede924ad
This will prevent the above error.